File: AspNetMvcUnleashed/0672329980_source/Chapter04/VB/MvcApplication1/MvcApplication1.Tests/Helpers/ProductHelperTest.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports MvcFakes

<TestClass()> _
Public Class ProductHelperTest
    <TestMethod()> _
    Public Sub ContainsHtmlRow()
        ' Arrange products
        Dim products = New List(Of Product)()
        products.Add(Product.CreateProduct(-1, "Laptop", "A laptop", 878.23D))
        products.Add(Product.CreateProduct(-1, "Telescope", "A telescape", 200.19D))

        ' Arrange HTML helper
        Dim helper = New FakeHtmlHelper()
        helper.ViewData.Model = products

        ' Act
        Dim result = ProductHelper.ProductList(helper)

        ' Assert
        StringAssert.Contains(result, "<td>Laptop</td><td>$878.23</td>")
    End Sub
End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter04/CS/MvcApplication1/MvcApplication1.Tests/Helpers/ProductHelperTest.cs
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Helpers;
using MvcApplication1.Models;
using MvcFakes;

namespace MvcApplication1.Tests.Helpers
{
    [TestClass]
    public class ProductHelperTest
    {
        [TestMethod]
        public void ContainsHtmlRow()
        {
            // Arrange products
            var products = new List<Product>();
            products.Add(Product.CreateProduct(-1, "Laptop", "A laptop", 878.23m));
            products.Add(Product.CreateProduct(-1, "Telescope", "A telescape", 200.19m));

            // Arrange HTML helper
            var helper = new FakeHtmlHelper();
            helper.ViewData.Model = products;

            // Act
            var result = ProductHelper.ProductList(helper);
        
            // Assert
            StringAssert.Contains(result, "<td>Laptop</td><td>$878.23</td>");
        }
    }
}
